cmd/collector/cmd/example.go

```diff
@@ -42,37 +42,22 @@ var exampleCmd = &cobra.Command{
 		collector.RegisterDocumentCollector(mc, mc.Type())
 
 		// Collect
-		docChan, errChan, numCollectors, err := collector.Collect(ctx)
-		if err != nil {
-			logrus.Error(err)
-			os.Exit(1)
-		}
-
-		collectorsDone := 0
-		for collectorsDone < numCollectors {
-			select {
-			case d := <-docChan:
-				emit(d)
-				logrus.Infof("emitted document: %+v", d)
-			case err = <-errChan:
-				if err != nil {
-					logrus.Errorf("collector ended with error: %v", err)
-				} else {
-					logrus.Info("collector ended gracefully")
-				}
-				collectorsDone += 1
+		errHandler := func(err error) bool {
+			if err == nil {
+				logrus.Info("collector ended gracefully")
+				return true
 			}
+			logrus.Errorf("collector ended with error: %v", err)
+			return false
 		}
-
-		// Drain anything left in document channel
```

Comment:
Correct me if I am wrong but don't we still have to drain the channel?

Reply:
yep we do - this chunk should be present in collector.go